cdc/processor/pipeline/table_actor.go

```diff
@@ -0,0 +1,457 @@
+		log.Error("sorter fails to start",
+			zap.String("tableName", t.tableName),
+			zap.Int64("tableID", t.tableID),
+			zap.Error(err))
+		return err
+	}
+	t.sortNode = sorterNode
+
+	pullerNode := newPullerNode(t.tableID, t.replicaInfo, t.tableName, t.changefeedVars.ID)
+	pullerActorNodeContext := NewContext(sdtTableContext,
+		t.tableName,
+		t.globalVars.TableActorSystem.Router(),
+		t.actorID, t.changefeedVars, t.globalVars, t.reportErr)
+	if err := pullerNode.InitWithWaitGroup(pullerActorNodeContext, t.wg, true, sorterNode); err != nil {
+		log.Error("puller fails to start",
+			zap.String("tableName", t.tableName),
+			zap.Int64("tableID", t.tableID),
+			zap.Error(err))
+		return err
+	}
+	t.pullerNode = pullerNode
+
+	var messageFetchFunc AsyncMessageHolderFunc
+	var messageProcessFunc AsyncMessageProcessorFunc
+
+	var cyclicActorNodeContext *cyclicNodeContext
+	if t.cyclicEnabled {
+		cyclicNode := newCyclicMarkNode(t.markTableID)
+		cyclicActorNodeContext = NewCyclicNodeContext(
+			NewContext(sdtTableContext, t.tableName,
+				t.globalVars.TableActorSystem.Router(),
+				t.actorID, t.changefeedVars,
+				t.globalVars, t.reportErr))
+		if err := cyclicNode.Init(cyclicActorNodeContext); err != nil {
+			log.Error("sink fails to start",
+				zap.String("tableName", t.tableName),
+				zap.Int64("tableID", t.tableID),
+				zap.Error(err))
+			return err
+		}
+		// construct cyclic actor node if it's enabled, it gets message from sortNode
+		messageFetchFunc = func() *pipeline.Message { return sortActorNodeContext.tryGetProcessedMessage() }
+		messageProcessFunc = func(ctx context.Context, msg pipeline.Message) (bool, error) {
+			return cyclicNode.TryHandleDataMessage(cyclicActorNodeContext, msg)
+		}
+		t.nodes = append(t.nodes, NewActorNode(messageFetchFunc, messageProcessFunc))
+	}
+
+	actorSinkNode := newSinkNode(t.tableID, t.tableSink,
+		t.replicaInfo.StartTs,
+		t.targetTs, flowController)
+	actorSinkNode.InitWithReplicaConfig(true, t.replicConfig)
+	t.sinkNode = actorSinkNode
+
+	// construct sink actor node, it gets message from sortNode or cyclicNode
+	if t.cyclicEnabled {
+		messageFetchFunc = func() *pipeline.Message { return cyclicActorNodeContext.tryGetProcessedMessage() }
+	} else {
+		messageFetchFunc = func() *pipeline.Message { return sortActorNodeContext.tryGetProcessedMessage() }
+	}
+	messageProcessFunc = func(ctx context.Context, msg pipeline.Message) (bool, error) {
+		return actorSinkNode.HandleMessage(sdtTableContext, msg)
+	}
+	t.nodes = append(t.nodes, NewActorNode(messageFetchFunc, messageProcessFunc))
+
+	t.started = true
+	log.Info("table actor is started",
+		zap.String("tableName", t.tableName),
+		zap.Int64("tableID", t.tableID))
+	return nil
+}
+
+// stop will set this table actor state to stopped and releases all goroutines spawned
+// from this table actor
+func (t *tableActor) stop(err error) {
+	t.stopLock.Lock()
```

Comment:
It is better to add a log here like `log.Info("table actor begin to stop....")`. I think it will be helpful for us to do some investigation when something went wrong here.

Reply:
done